components/MarkdownEditor.tsx

```diff
@@ -0,0 +1,248 @@
+          ></path>
+        </svg>
+      </Tooltip>
+    ),
+    execute: (state: TextState, api: TextAreaTextApi) => {
+      const linkText = state.selectedText ? state.selectedText : "";
+      const selection = {
+        start: state.selection.start,
+        end: state.selection.end,
+        linkText: linkText,
+      };
+      openLinkPrompt(selection);
+    },
+  };
+
+  const onConfirmURLPrompt = (url: string) => {
+    const { start, end } = selectionInfo;
+    const before = value.substring(0, start);
+    const transformedLink = transformLink(url);
+    const after = value.substring(end);
+    const modifyText = `${before}[${
+      selectionInfo.linkText || linkText
+    }](${transformedLink})${after}`;
+    onChange(modifyText);
+    setValue(modifyText);
+    onCloseURLPrompt();
+  };
+
+  const onCloseURLPrompt = () => {
+    setLinkText("");
+    setUrl("");
+    setIsOpenUrlPrompt(false);
+  };
+
+  return (
+    <div id={id} data-color-mode="light">
+      <Scrim
+        open={isOpenUrlPrompt}
+        onClose={onCloseURLPrompt}
+        isDismissable
+        onKeyPress={(e) => {
+          if (e.key === "Enter") {
+            onConfirmURLPrompt(url);
+          }
+        }}
+        style={{
+          overflow: "hidden",
+          backgroundColor: "white",
+          position: "absolute",
+          bottom: 0,
+          left: 0,
+          zIndex: 1000,
+          display: "block",
+          width: "100%",
+        }}
+      >
+        <div className={styles.scrimWrapper}>
+          <div
+            className={styles.scrimHeaderWrapper}
+            style={{ marginBottom: 16 }}
+          >
+            <div className={styles.scrimTitle}>Add link</div>
+          </div>
+          <TextField
+            autoFocus={!selectionInfo.linkText}
+            label={"Text"}
+            disabled={!!selectionInfo.linkText}
+            variant={"default"}
+            defaultValue={selectionInfo.linkText || linkText}
```

Comment:
selectionInfo.linkText is the selectedText in the textarea.

Local state "linkText" was for saving the link text from the "Add link" form temporarily. I had it like this to be able to disable the input field if you had already selected some text before adding a link. I now removed it so that it will never be disabled but it will focus different TextField depending on what text you had selected before adding a link.

It now ended ut being a more flexible solution by never disabling the "Text" input field in the "Add link" form💪